samples/generics/iterator.jakt

```diff
@@ -0,0 +1,18 @@
+/// Expect:
+/// - output: "1\n2\n3\nA\nB\nC\n"
+
+function foo<T>(anon xs: T) {
```

Comment:
:thinking: I wonder if we should require `foo<T requires(Iterable<U>)>` to allow iteration?
This is fine for now since that requirement doesn't actually fix the issue, and we still have template-y generics